circom/tests/loops/inner_conditional_10.circom

```diff
@@ -12,6 +12,9 @@ template Poseidon() {
 
     component sigmaF[2];
 
+    // NOTE: When processing the loop, the statements indexed with 'k' are determined
+    //  NOT safe to move into a new function since 'k' is unknown. That results in
+    //  the loop unrolling in place.
```

Comment:
Technically yes. The algo is a bit naive about this since it's assigned within the body of the loop. Already have a ticket in JIRA for it: https://veridise.atlassian.net/browse/VAN-679